### ISSUE TITLE
Broken primaries on FxOS

### ISSUE THREAD

**Issue by jaredhirsch** [mozilla/persona#4144]
Summary: Right now, it is not possible to log into Persona on FxOS using a primary IdP.

Looks like CSP inline script errors, but this might not actually be the cause?

Jed snagged a bit of `adb logcat` output while investigating an unrelated Gaia bug. The output mentions CSP errors and seems relevant (the snippet below seems most relevant, the rest is in [a pastebin](https://pastebin.mozilla.org/5333739)).

E/GeckoConsole(17036): [JavaScript Warning: "The X-Content-Security-Policy and X-Content-Security-Report-Only headers will be deprecated in the future. Please use the Content-Security-Policy and Content-Security-Report-Only headers with CSP spec compliant syntax instead." {file: "https://login.mozilla.org/provision" line: 0}]
I/Gecko   (17036): nsDOMIdentity ({1f4b18d7-231f-47e5-8771-33398706c6c3}): init was called from https://login.mozilla.org/provision
E/GeckoConsole(17036): [JavaScript Warning: "SyntaxError: Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead" {file: "https://login.mozilla.org/js/lib/jquery.min.js" line: 1}]
E/GeckoConsole(17036): [JavaScript Warning: "Error: https://login.mozilla.org/js/lib/jquery.min.js is being assigned a //# sourceMappingURL, but already has one"]
I/Gecko   (17036): nsDOMIdentity ({1f4b18d7-231f-47e5-8771-33398706c6c3}): beginProvisioning
I/Gecko   (17036): nsDOMIdentity ({1f4b18d7-231f-47e5-8771-33398706c6c3}): DOMIdentityMessage: {"errors":[],"id":"{1f4b18d7-231f-47e5-8771-33398706c6c3}","origin":"https://login.mozilla.org","appStatus":0,"audience":"https://login.mozilla.org"}
E/GeckoConsole(16573): [JavaScript Error: "getService called before newContext for {1f4b18d7-231f-47e5-8771-33398706c6c3}" {file: "resource://gre/modules/DOMIdentity.jsm" line: 201}]
**E/GeckoConsole(17036): [JavaScript Warning: "Content Security Policy: The page's settings blocked the loading of a resource: An attempt to execute inline scripts has been blocked" {file: "https://login.mozilla.org/provision" line: 0 column: 0 source: "onsubmit attribute on DIV element"}]
E/GeckoConsole(17036): [JavaScript Warning: "Content Security Policy: The page's settings blocked the loading of a resource: An attempt to execute inline scripts has been blocked" {file: "https://login.mozilla.org/provision" line: 0 column: 0 source: "onchange attribute on DIV element"}]
E/GeckoConsole(17036): [JavaScript Warning: "Content Security Policy: The page's settings blocked the loading of a resource: An attempt to execute inline scripts has been blocked" {file: "https://login.mozilla.org/provision" line: 0 column: 0 source: "onsubmit attribute on DIV element"}]
E/GeckoConsole(17036): [JavaScript Warning: "Content Security Policy: The page's settings blocked the loading of a resource: An attempt to execute inline scripts has been blocked" {file: "https://login.mozilla.org/provision" line: 0 column: 0 source: "onchange attribute on DIV element"}]**

What's bizarre about this error is that there's no inline onchange or onsubmit handlers inside the provision page--the mozidp provision page doesn't have [any div els at all](https://github.com/mozilla/vinz-clortho/blob/master/server/views/provision.ejs).

**Comment by jaredhirsch** [comment]
I didn't mention it in the bug description, but this seems possibly related to the addition of CSP headers in #4012

**Comment by jaredhirsch** [comment]
@seanmonstar Any thoughts on CSP config settings or maybe known browser bugs that might have led to the weird errors above?

**Comment by jedp** [comment]
I've filed a bugzilla bug to track this for FirefoxOS:

https://bugzilla.mozilla.org/show_bug.cgi?id=1020020

**Comment by seanmonstar** [comment]
We did add CSP headers to login.persona.org that prevent inline scripts... The provision page is being loaded in an iframe on login.persona.org. That sounds like incorrect behavior that the CSP is being applied to the iframe, since iframes. Also, it's not happening on desktop browsers, so I'm thinking it'd something conflicting with the way FxOS ropes in Persona bits.

**Comment by fmarier** [comment]
I've been unable to log into Persona using my Mozilla email (Firefox Desktop 29) for a few weeks now, maybe it's related...

**Comment by seanmonstar** [comment]
I've been able to login to id.etherpad.mozilla.org, which uses my IdP-backed mozilla.com email.

**Comment by jedp** [comment]
@seanmonstar it could well be something FxOS-specific.  The error messages are confusing: "onsubmit attribute on DIV element"?  What DIV element?

**Comment by callahad** [comment]
Those CSP errors are a red herring.
1. They are from the MoCo/MoFo IdP (login.mozilla.org, not login.persona.org). That IdP has not changed in ages.
2. Those CSP errors appear in my browser console, despite everything working just perfectly.

**Comment by callahad** [comment]
I'm especially interested in whether or not mockmyid.com and gmail.com addresses work. Do they?

**Comment by jedp** [comment]
When I get home, I'll try with a last-year's build of firefoxos - just to be damn sure this isn't a platform bug

**Comment by jedp** [comment]
The same problem manifests with a nice old 2013-10-09 build of FxOS 1.1 for Unagi.  (Remember the dear old unagi?)  Trying to provision with a primary (my @mozilla.com address) hangs like so:

![2014-06-05-12-07-07](https://cloud.githubusercontent.com/assets/532953/3190220/30e6b710-eccc-11e3-93d2-548148cf2fd0.png)

adb logcat doesn't show me anything interesting on the unagi (those CSP warnings are not present).  It just looks like it tries and gives up after a while:

I/Gecko   (  783): nsDOMIdentity ({3d1cd693-2908-405a-8a11-b170ccc064ce}): init was called from https://login.mozilla.org/provision
I/Gecko   (  783): nsDOMIdentity ({3d1cd693-2908-405a-8a11-b170ccc064ce}): beginProvisioning
I/ONCRPC  (  115): rpc_handle_rpc_call: for Xid: 148, Prog: 31000000, Vers: fc37ad5c, Proc: 00000012
I/ONCRPC  (  115): rpc_handle_rpc_call: Find Status: 0 Xid: 148
I/ONCRPC  (  115): oncrpc_proxy_handle_cmd_rpc_call: Dispatching xid: 148
I/ONCRPC  (  115): rpc_handle_rpc_call: for Xid: 149, Prog: 31000003, Vers: cf0d42a3, Proc: 00000003
I/ONCRPC  (  115): oncrpc_proxy_handle_cmd_rpc_call: Dispatching xid: 149
I/ONCRPC  (  115): rpc_handle_rpc_call: Find Status: 0 Xid: 149
I/ONCRPC  (  115): Setup RPC Call for task 4017d420
I/ONCRPC  (  115): oncrpc_xdr_call_msg_start: Prog: 30000000, Ver: ddc9bd91, Proc: 000000b0
I/ONCRPC  (  115): xdr_std_msg_send_call: Sent Xid: b7, Prog: 30000000, Ver: ddc9bd91, Proc: 000000b0
I/ONCRPC  (  115): xdr_std_msg_send_call: Received Reply Xid: b7, Prog: 30000000, Ver: ddc9bd91, Proc: 000000b0
I/ONCRPC  (  115): rpc_handle_rpc_call: for Xid: 14a, Prog: 31000000, Vers: fc37ad5c, Proc: 00000004
I/ONCRPC  (  115): rpc_handle_rpc_call: Find Status: 0 Xid: 14a
I/ONCRPC  (  115): oncrpc_proxy_handle_cmd_rpc_call: Dispatching xid: 14a
I/ONCRPC  (  115): rpc_handle_rpc_call: for Xid: 14b, Prog: 31000000, Vers: fc37ad5c, Proc: 0000000f
I/ONCRPC  (  115): rpc_handle_rpc_call: Find Status: 0 Xid: 14b
I/ONCRPC  (  115): oncrpc_proxy_handle_cmd_rpc_call: Dispatching xid: 14b
I/ONCRPC  (  115): rpc_handle_rpc_call: for Xid: 14c, Prog: 31000000, Vers: fc37ad5c, Proc: 00000012
I/ONCRPC  (  115): rpc_handle_rpc_call: Find Status: 0 Xid: 14c
I/ONCRPC  (  115): oncrpc_proxy_handle_cmd_rpc_call: Dispatching xid: 14c
E/rmt_storage(  136): write_log_file: [MyTag]open: fd=-1
E/rmt_storage(  136): block rmt_storage client thread
I/ONCRPC  (  115): rpc_handle_rpc_call: for Xid: 154, Prog: 31000000, Vers: fc37ad5c, Proc: 00000012
I/ONCRPC  (  115): rpc_handle_rpc_call: Find Status: 0 Xid: 154
I/ONCRPC  (  115): oncrpc_proxy_handle_cmd_rpc_call: Dispatching xid: 154
I/Gecko   (  783): nsDOMIdentity ({3d1cd693-2908-405a-8a11-b170ccc064ce}): nsDOMIdentity uninit()

**Comment by jaredhirsch** [comment]
@callahad mockmyid and gmail addresses do not work, same symptoms: blank provisioning page, followed by a provisioning error. Unfortunately the trusted UI iframe on device cannot be inspected by FxOS devtools, so I can't actually peek at the provision page to see if it's loaded without any content, loaded but scripts stripped out, or not loaded at all :-\

@jedp OK, so it sounds like probably a server thing, then.

Whenever awsboxes are fixed (I'm swamped with fxos stuff, haven't kept up with this repo), maybe we can fire up an ephemeral instance with the CSP headers disabled and see if that fixes things.

**Comment by callahad** [comment]
OK, this is pretty strange.

Here's what I can see using the 1.3 Simulator:

| **RP** | **API** | **Primary Works?** | **Secondary Works?** | **Notes** |
| --- | --- | --- | --- | --- |
| Marketplace | Observer + forceIssuer + allowUnverified | :interrobang: | :white_check_mark: | Forces issuer to login.persona.org. |
| 123Done | Observer | :x: | :white_check_mark: | Primaries: Times out when attempting to provision |

**Additional Notes:**

When watching the server logs for MockMyID, I don't see FxOS request the provisioning page _at all_.

After establishing a fallback password for MockMyID (sign into Marketplace on Desktop), then attempting to sign into 123Done on the simulator, I correctly see the "Hooray! You've been upgraded to a primary screen!" When I click "Next", _the Primary's authentication page **does** load_. However, nothing happens after that. (Verified for both MockMyID and Gmail Bridge.)

Oddly enough, it _is_ possible to sign into Marketplace with a primary. First, sign into Marketplace on Desktop with a primary, and verify your email. Then attempt to sign into Marketplace on the Simulator. It should work.

Of course, the normal path for primary users on Marketplace fails. They are correctly prompted to create a password, but upon hitting submit, they see either the Persona "Server too busy" error, or if the dialog closes, a Marketplace toast indicating that "a server error was encountered." In the console, Marketplace suggests that this is related to failed verification.

**Comment by callahad** [comment]
So, it seems like the root issue is that _FxOS never requests the IdP's provisioning page,_ even when triggered by `navigator.id.completeAuthentication()` on the IdP's authentication page.

The fact that the latter fails for the Gmail bridge, [which has a strict CSP](https://github.com/mozilla/persona-gmail-bridge/blob/master/lib/csp.js), is a bit mind boggling.

**Comment by callahad** [comment]
So this suggests that the issue is with the normal Persona popup window (/sign_in), when it tries to create an iframe to the provisioning page.

**Comment by jaredhirsch** [comment]
It looks like it _is_ possible to examine the trusted UI iframe contents with the simulator, since it doesn't open `<iframe remote>` iframes in a separate process, but inside the main process.

Now I just have to figure out how to build the Simulator and I can get some useful info.

**Comment by jaredhirsch** [comment]
Gah, didn't mean to close.

**Comment by jaredhirsch** [comment]
Lowering to four stars, since forceIssuer avoids this problem, Marketplace uses forceIssuer, and they're the only critically-important RP on the phone.

**Comment by callahad** [comment]
Re-raising to 5-Star (sorry!) -- forceIssuer does _not_ avoid this problem.

It's impossible to sign into Marketplace with a primary, unless you use a crazy workaround: First sign into Marketplace on Desktop and then immediately sign into Marketplace on FxOS. If you sign into any other RPs, you have to repeat the Desktop -> FxOS dance.

**Comment by callahad** [comment]
(Sorry; upon re-reading, my text above wasn't very clear.)

**Comment by shane-tomlinson** [comment]
@jedp, IIRC we talked about CSP being a potential problem when we first integrated Persona into FxOS. I brought up that there was a bug in FxOS with CSP not being applied, and that was the only reason why we were able to work with primaries. Ben talked with the platform and gaia and told them we were depending on the behavior, and the outcome was along the lines of "they have agreed not to screw us by changing the CSP rules". Have they changed the CSP rules?

**Comment by shane-tomlinson** [comment]
Ah, the specific problem now that I remember is that CSP rules were being correctly applied to the IFRAME containing Persona, but _not_ to the IFRAMEs embedded in Persona content. If they fixed that issue, hosed. I could be completely mis-remembering this.

**Comment by shane-tomlinson** [comment]
Link to #3060 and #3566

**Comment by shane-tomlinson** [comment]
Also link to https://bugzilla.mozilla.org/show_bug.cgi?id=456957, https://bugzilla.mozilla.org/show_bug.cgi?id=856548

**Comment by callahad** [comment]
So we need to drop CSPs on FxOS, or perhaps more simply, altogether?

**Comment by shane-tomlinson** [comment]
@callahad - I believe the problem may be FxOS CSP rules, not ours, but I am not positive.

**Comment by callahad** [comment]
1. This wasn't reported until two weeks after our production push, which
   suggests an issue independent of Persona.
2. The 1.3 simulator (build 20140331164005) exhibits the problem, which
   suggests an issue with Persona.

Given those two conditions, I'm inclined to believe that it's a problem
with us.

**Comment by jedp** [comment]
It may have gone undetected because most of the FxOS testing takes place using personatestuser, which provisions secondary accounts with persona.org directly.  There may not be many people signing into sites on FirefoxOS using persona with their primary address.  This came up as a result of someone trying to get into bugzilla with a @mozilla.com address.

@shane-tomlinson CSP could have changed on FxOS, causing additional problems, but in my comment above from June 5, I reproduced the problem with an October 2013 build of FxOS on unagi.  So I feel it's independent of the FxOS build.

**Comment by callahad** [comment]
This should be fixed in stage later today, targeting a production fix on Monday.

**Comment by callahad** [comment]
Bad news. Reopening. There's a pre-prod stack (should go prod early tomorrow) that does not send CSP headers.

It still exhibits the same issue: requests are _never_ made to the Primary, as per this screenshot:

![screen shot 2014-06-17 at 19 08 18](https://cloud.githubusercontent.com/assets/24193/3308390/3c4ec3a8-f67d-11e3-8ff3-26e16b3811d2.png)

@jedp, @6a68, @shane-tomlinson: I need your help. I don't have a device, nor are my FxOS skills up to snuff. I'm not able to debug this further without assistance.

**Comment by shane-tomlinson** [comment]
@callahad - We need to find out if FxOS folks have fixed the CSP problem of rules not being extended to child IFRAMEs even though they should be.

> @jedp, IIRC we talked about CSP being a potential problem when we first integrated Persona into FxOS. I brought up that there was a bug in FxOS with CSP not being applied, and that was the only reason why we were able to work with primaries. Ben talked with the platform and gaia and told them we were depending on the behavior, and the outcome was along the lines of "they have agreed not to screw us by changing the CSP rules". Have they changed the CSP rules?

**Comment by shane-tomlinson** [comment]
Digging into mxr: http://mxr.mozilla.org/mozilla-central/source/b2g/app/b2g.js#396

> // Default Content Security Policy to apply to privileged and certified apps
> 397 pref("security.apps.privileged.CSP.default", "default-src *; script-src 'self'; object-src 'none'; style-src 'self' 'unsafe-inline'");
> 398 // If you change this CSP, make sure to update the fast path in nsCSPService.cpp
> 399 pref("security.apps.certified.CSP.default", "default-src *; script-src 'self'; object-src 'none'; style-src 'self' 'unsafe-inline'");

Trying to find out more.

**Comment by shane-tomlinson** [comment]
Huh, I can repro on Fx 1.3 prerelease from a long long time ago. I may be chasing dead ends.

**Comment by shane-tomlinson** [comment]
mozilla/fxa-content-server#1241 is to upgrade jQuery to 1.11.1 because the 1.10.2 branch caused CSP violations in the way jQuery did some feature detection. Could be related.

**Comment by callahad** [comment]
Prod is updated. It no longer sends CSP headers. The bug still exists.

**Comment by jaredhirsch** [comment]
@shane-tomlinson Yeah, Jed was able to repro with an ancient device as well (see his [earlier comment](https://github.com/mozilla/persona/issues/4144#issuecomment-45240587) in this bug). Seems like we can rule out FxOS platform changes.

About that jquery bug, the bug report is [here](http://bugs.jquery.com/ticket/14422) and [here's](https://github.com/jquery/jquery/commit/9e3d0f3109756ec8b6166ff60f0d495b8f1d6aca) the git commit. Doing a [google search](https://www.google.com/?gws_rd=ssl#q=site:bugs.jquery.com+csp) against the jquery bugs site, there's no shortage of similar bugs. CSP is maddeningly fiddly.

Here's a nice collection of CSP violations, along with suggestions for fixes: http://www.cspplayground.com/violation_examples

I'd suggest we stand up a staging stack pointed at train-10.09 and make sure we can verify logins on FxOS really do work with that code. Then we'll have a basis to git bisect further.

I have an idea about how to debug more deeply. It seems like we will need to stand up an awsbox, pin its DNS name in a desktop machine's /etc/hosts (since the personatest.org domain is still borked), and hopefully that'll make it possible to use the FxOS simulator to use the awsbox for CSP errors. If this is possible, it'll just be a matter of git bisecting with the awsbox. The trickiest part will be ensuring that the awsbox sends the same CSP headers as production.

I don't have the time right now, but I can try tonight. Anybody else is also welcome to try; since jrgm's awsbox fix landed, we can actually provision boxes, though DNS naming is still busted. (Maybe we should file a bug to get persona-test.org or something? Thoughts?)

@jedp how can I configure FxOS to use a different Persona server? What are the prefs to set?

**Comment by callahad** [comment]
> The trickiest part will be ensuring that the awsbox sends the same CSP headers as production.

Production does not send CSP headers. The bug still exists.

Switching to an awsbox and bisecting sounds like a good idea. I can try to dig in later tonight, as well.

**Comment by callahad** [comment]
@edmoz reports seeing "E/GeckoConsole( 9823): [JavaScript Error: "content.wrappedJSObject.BrowserID is undefined" {file: "chrome://b2g/content/identity.js" line: 103}]" on a Flame running B2G master (~2.0)

**Comment by edwindotcom** [comment]
buri with 1.3 master head - hotmail (secondary)
- sign in as expected

buri with 1.3 master head - mockmyid

on device: provisioning error screen

```
and this in adb logcat with identity debug = true
https://edwong.pastebin.mozilla.org/5431486
E/GeckoConsole(  954): Identity injected identity.js: doInternalRequest: true
E/GeckoConsole(  954): Security Error: Content at https://mockmyid.com/ may not load data from https://firefoxos.persona.org/sign_in#NATIVE.
I/Gecko   (  954): nsDOMIdentity ({6226ef27-cb25-4b30-ac5c-2ee7c03506c9}): init was called from https://mockmyid.com/browserid/provision.html
I/Gecko   (  954): nsDOMIdentity ({6226ef27-cb25-4b30-ac5c-2ee7c03506c9}): beginProvisioning
I/Gecko   (  954): nsDOMIdentity ({6226ef27-cb25-4b30-ac5c-2ee7c03506c9}): nsDOMIdentity uninit()
```

buri with 1.3 master head - gmail

on device: provisioning error screen

```
I/Gecko   ( 1031): Identity injected identity.js: doInternalRequest: true
E/GeckoConsole( 1031): Identity injected identity.js: doInternalRequest: true
E/GeckoConsole( 1031): Security Error: Content at https://gmail.login.persona.org/ may not load data from https://firefoxos.persona.org/sign_in#NATIVE.
E/GeckoConsole( 1031): [JavaScript Warning: "The X-Content-Security-Policy and X-Content-Security-Report-Only headers will be deprecated in the future. Please use the Content-Security-Policy and Content-Security-Report-Only headers with CSP spec compliant syntax instead." {file: "https://gmail.login.persona.org/provision" line: 0}]
I/Gecko   ( 1031): nsDOMIdentity ({9cb2d297-7ac7-40b7-a2ae-89c92571b8f0}): init was called from https://gmail.login.persona.org/provision
I/Gecko   ( 1031): nsDOMIdentity ({9cb2d297-7ac7-40b7-a2ae-89c92571b8f0}): beginProvisioning
```

buri with 1.3 master head - mozilla.com

on device: provisioning error screen

```
I/Gecko   ( 1304): Identity injected identity.js: doInternalRequest: true
E/GeckoConsole( 1304): Identity injected identity.js: doInternalRequest: true
E/GeckoConsole( 1304): Security Error: Content at https://login.mozilla.org/ may not load data from https://firefoxos.persona.org/sign_in#NATIVE.
E/GeckoConsole( 1304): [JavaScript Warning: "The X-Content-Security-Policy and X-Content-Security-Report-Only headers will be deprecated in the future. Please use the Content-Security-Policy and Content-Security-Report-Only headers with CSP spec compliant syntax instead." {file: "https://login.mozilla.org/provision" line: 0}]
I/Gecko   ( 1304): nsDOMIdentity ({37478ebf-6a7f-4c0a-a4d0-2b51e3d3a9bc}): init was called from https://login.mozilla.org/provision
E/GeckoConsole( 1304): [JavaScript Warning: "SyntaxError: Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead" {file: "https://login.mozilla.org/js/lib/jquery.min.js" line: 1}]
E/GeckoConsole( 1304): [JavaScript Warning: "Error: https://login.mozilla.org/js/lib/jquery.min.js is being assigned a //# sourceMappingURL, but already has one"]
I/Gecko   ( 1304): nsDOMIdentity ({37478ebf-6a7f-4c0a-a4d0-2b51e3d3a9bc}): beginProvisioning
E/GeckoConsole( 1304): [JavaScript Warning: "Content Security Policy: The page's settings blocked the loading of a resource: An attempt to execute inline scripts has been blocked" {file: "https://login.mozilla.org/provision" line: 0 column: 0 source: "onsubmit attribute on DIV element"}]
```

**Comment by callahad** [comment]
That looks like it:

`E/GeckoConsole(  954): Security Error: Content at https://mockmyid.com/ may not load data from https://firefoxos.persona.org/sign_in#NATIVE`

So... how do we fix that? And why has it only now become a problem?

**Comment by callahad** [comment]
The above error is misleading. We saw it much earlier on, and FxOS continued to work:

https://github.com/mozilla/persona/pull/3461#discussion_r4439836

**Comment by jaredhirsch** [comment]
Copying over my comment in #4154, summarizing Shane's excellent sleuthing:

It looks like this commit broke primaries on FxOS because it sets `navigator.id = navigator.mozId`, but then we try to add properties to navigator.id. In FxOS, `navigator.mozId` is immutable, and will silently fail if we try to modify it. This is a pretty good guess as to the cause of #4144.

**Comment by jaredhirsch** [comment]
@gene1wood any rough ETA on getting personatest back up and running? Maybe we should throw the #4154 branch on stage and see if it fixes the bug?

**Comment by gene1wood** [comment]
I was very close to transferring the domain so I could get the cert but then realized I need to get the current DNS zone information for the zones we're transferring so we can create them in mozilla DNS servers.

This _may_ not apply however for personatest.org because maybe it has no DNS settings? If so I'll have webops do just that domain on Monday.

For the rest I'll work on getting in touch with Lloyd to get the info (I emailed him asking for it or access to his DNS providers but haven't heard back yet)

So if personatest.org has no DNS zone settings, then I can probably get the new cert by Monday COB

**Comment by jrgm** [comment]
@gene1wood - so personatest.org name servers are route53 NS under our AWS account control.

**Comment by jaredhirsch** [comment]
Awesome sauce! I'll take #4154 for a spin once we've got the cert back in working order, and hopefully our long nightmare ends there. Thanks again for driving the cleanup efforts, Gene :beers:

**Comment by gene1wood** [comment]
Damn. I've been conflating personatestuser.org and personatest.org. Lloyd hasn't gotten us the transfer code for personatest.org because I asked for the other one.

I've emailed him and will call him when I'm in the office.

**Comment by gene1wood** [comment]
Ok, just talked to @lloyd (was just heading into a movie) and he'll get us the transfer code for personatest.org after it's out (he said he was doing some work this evening anyhow). Once I get that mid-day I'll request the transfer from webops

**Comment by jaredhirsch** [comment]
@gene1wood any updates?

**Comment by jaredhirsch** [comment]
The quickest way forward, I think, is to put this branch on stage (mozilla/identity-ops#160) and see if it fixes the problem on device. Feature complete is creeping closer, and we need to get this fixed.

**Comment by gene1wood** [comment]
This fix is present in staging now.

**Comment by edwindotcom** [comment]
Tests pass on mac Fx and B2G (pushed stage pref) with beta.123done.org:
mockmyid.com, gmail, yahoo, hotmail (secondary)

**Comment by edwindotcom** [comment]
john is also going to take a look.  we should give the green light by monday.

**Comment by jaredhirsch** [comment]
Fantastic. Thanks, @edmoz!

**Comment by jaredhirsch** [comment]
@edmoz Just for planning purposes: I'm happy to merge that PR and cut a train once we're done verifying. We can do a release next week.

**Comment by karlht** [comment]
Testing MozIdP on stage ... @mostlygeek, https://login.allizom.org is not responding.  Do we know why?

**Comment by mostlygeek** [comment]
I have resurrected https://login.allizom.org. It wasn't getting any traffic for a long time so it's been down for about a month. It should be working now.

**Comment by edwindotcom** [comment]
![giphy](https://cloud.githubusercontent.com/assets/3698618/3604358/5d16bd44-0d1f-11e4-8f36-780f65b86dac.gif)

**Comment by karlht** [comment]
https://login.allizom.org is back up.  Prefs pushed, MozIdP works fine in staging.

**Comment by relud** [comment]
Does that mean we're ready to move this to prod?

**Comment by edwindotcom** [comment]
John is running auto tests.  well log bugzilla tix when ready

**Comment by gene1wood** [comment]
@edmoz we moved to using identity-ops issues instead of bugzilla tickets with the last few releases so we can link the config management code changes that lock in the versioning to the release ticket.

**Comment by jrgm** [comment]
I can only run qunit tests without a valid personatest.org cert, so they pass, and whatever.

**Comment by shane-tomlinson** [comment]
This is a silly question, what is the magic incantation to use login.allizom.org as the IdP from https://login.anosrep.org/

**Comment by callahad** [comment]
Use `login.allizom.org` as your email domain. If I recall correctly, it
talks to LDAP, so your normal username + password should work.

**Comment by edwindotcom** [comment]
sounds like we're a go for pushing stage to prod. closing this out. I'll log a identity-ops issue shortly

**Comment by jaredhirsch** [comment]
OK, recapping:
- we have a possible fix for this bug, #4154 
- ops put the fix on stage for testing
- looks like the test works
- next step is to merge the fix and cut a branch, following [our process for releasing code](https://github.com/mozilla/persona/wiki/How-to-Branch-The-Trains)

Updated: I've opened bug #4164 to branch the train. If @callahad gets to it before me, he can grab it, otherwise I'll cut a branch and email the list later today.

**Comment by jaredhirsch** [comment]
Update: see mozilla/identity-ops#168 to track progress of the deploy.
